### ISSUE TITLE
Update storybook-a11y-test package to support React 18

### ISSUE THREAD

**Issue by ryanwilsonperkin** [Shopify/quilt#2255]
`storybook-a11y-test` has a failing test when updating to React 18

```
can test a story › secondary has a11y errors

    expect(received).toHaveLength(expected)

    Expected length: 1
    Received length: 0
    Received array:  []

      44 |       storyIds: ['example-button--secondary'],
      45 |     });
    > 46 |     expect(result).toHaveLength(1);
         |                    ^
      47 |   });
      48 |
      49 |   it('a11y disabled is skipped', async () => {

      at Object.<anonymous> (tests/index.test.ts:46:20)
```
